flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskExecutorPartitionLifecycleTest.java

```diff
@@ -102,6 +126,115 @@ public void setup() {
 		rpc = new TestingRpcService();
 	}
 
+	@Test
+	public void testConnectionTerminationAfterExternalRelease() throws IOException, InterruptedException, ExecutionException, TimeoutException {
+		final JobMasterId jobMasterId = JobMasterId.generate();
+
+		final LibraryCacheManager libraryCacheManager = mock(LibraryCacheManager.class);
+		when(libraryCacheManager.getClassLoader(any(JobID.class))).thenReturn(ClassLoader.getSystemClassLoader());
+
+		final JobMasterGateway jobMasterGateway = mock(JobMasterGateway.class);
```

Comment:
Could use new `TestingJobMasterGatewayBuilder().build()` to avoid mock and seems not necessary to consider `jobMasterId`.